libass/ass.c

```diff
@@ -1522,17 +1522,29 @@ ASS_Track *ass_new_track(ASS_Library *library)
 
 int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
 {
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+    bool *features[] = {
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+        [ASS_FEATURE_BIDI_BRACKETS] = &track->parser_priv->bidi_brackets,
+#else
+        [ASS_FEATURE_BIDI_BRACKETS] = NULL,
```

Comment:
Hmm, can’t this `#else` clause be omitted entirely? Not in the first commit because the array needs at least one element? But it would seem sensible to omit such clauses entirely in the future, as soon as at least one always-present element is added (which happens in the second commit).

Reply:
Currently no, because ASS_FEATURE_BIDI_BRACKETS takes a space in the enum and will be iterated.

Reply:
Sure, but won’t the remainder of the array (any elements that aren’t explicitly designated) be zero-initialized?